File: 3. Stack/Stack Implementation With Fixed Array/FixedStack.java

import java.util.Scanner;

public class FixedStack {
    private static int[] stack = new int[5];
    private static int stackPointer = -1;
    private static int stackSize = 5;

    public static void main(String[] args) {
        Scanner myObj = new Scanner(System.in);

        int choice;
        System.out.println("Hello! Here You Can Perform Following Array Operations!");
        while (true) {
            System.out.println("1. Push(Insert)\n2. Pop(Delete)\n3. Peek(See Last Element)\n4. Is Empty(Checks weather Stack is Empty)\n5. Is Full(Checks weather Stack is Full)\n6. Exit");
            System.out.print("Enter Your Choice:- ");
            choice = myObj.nextInt();

            switch (choice) {
                case 1:
                System.out.println("~ FixedStack.java:25 -> stackPointer: " + stackPointer);
                    if (stackPointer + 1 == stackSize) {
                        System.out.println("\n~~Stack is Full~~\n");
                        break;
                    }
                    stackPointer++;
                    System.out.print("\nEnter Your Number:- ");
                    int val = myObj.nextInt();
                    
                    stack[stackPointer] = val;
                    System.out.format("\nInsertion Successfull:- %d\n", val);
                    break;
                case 2:
                    if (stackPointer == -1) {
                        System.out.println("\n~~Stack is Empty~~\n");
                        break;
                    }
                    System.out.format("\nDeletion Successfull:- %d\n", stack[stackPointer]);
                    stackPointer--;
                    break;
                case 3:
                    if (stackPointer == -1) {
                        System.out.println("\n~~Stack is Empty~~\n");
                    } else {
                        System.out.format("\nData Element:- %d\n", stack[stackPointer]);
                    }
                    break;
                case 4:
                    if (stackPointer == -1) {
                        System.out.println("\n~~Stack is Empty~~\n");
                    } else {
                        System.out.println("\n~~Stack is Not Empty~~\n");
                    }
                    break;
                case 5:
                    if (stackPointer + 1 == stackSize) {
                        System.out.println("\n~~Stack is Full~~\n");
                    } else {
                        System.out.println("\n~~Stack is Not Full~~\n");
                    }
                    break;
                case 6:
                    System.out.println("\n~~Thanks for Using! Have a Great Day!~~\n");
                    myObj.close();
                    return;
                default:
                    System.out.println("\n~~Please Enter a Valid Input!~~\n");
            }
        }
    }
}

File: 3. Stack/Stack Implementation With Fixed Array/fixedStack.cpp
#include <iostream>
using namespace std;

int stack[10], stackPointer = -1, stackSize = 10;

void push() {
    if (stackPointer + 1 == stackSize) {
        cout << "\n~~ Stack is Full~~\n";
        return;
    }

    int val;
    cout <<"\nEnter Value:- ";
    cin >> val;
    stackPointer++;
    stack[stackPointer] = val;
    cout <<"Insertion Successfull. Value:- " << val << "\n";
}

void pop() {
    if (stackPointer == -1) {
        cout <<"\n~~ Stack is Empty~~\n";
        return;
    }

    cout <<"\nRemoved Value: " << stack[stackPointer] << "\n";
    stackPointer--;
}

int main() {
    int choice;
    cout << "Hello! Here You Can Perform Following Stack Operation!";
    while (1) {
        cout <<"\n1. Push(Insert)\n2. Pop(Delete)\n3. Peek(See Last Element)\n4. Is Empty(Checks weather Stack is Empty)\n5. Is Full(Checks weather Stack is Full)\n6. Exit\nEnter Your Choice:- ";
        cin >> choice;

        switch (choice) {
        case 1:
            push();
            break;
        case 2:
            pop();
            break;
        case 3:
            if (stackPointer == -1) {
                cout <<"\n~~ Stack is Empty~~\n";
            } else {
                cout <<"\nTop Element:- " << stack[stackPointer] << "\n";
            }
            break;
        case 4:
            if (stackPointer == -1) {
                cout <<"\n~~ Stack is Empty~~\n";
            } else {
                cout <<"\n~~ Stack is Not Empty~~\n";
            }
            break;
        case 5:
            if (stackPointer + 1 == stackSize) {
                cout <<"\n~~ Stack is Full~~\n";
            } else {
                cout <<"\n~~ Stack is Not Full~~\n";
            }
            break;
        case 6:
            cout <<"\n~~Thanks For Using! Have a Great Day!~~\n";
            return 0;
        default:
            cout <<"\n~~Please Enter a Vaild Input!~~\n";
            break;
        }
    }
    return 0;
}